### PR TITLE
adding check for sub cert lints

### PR DESCRIPTION
**Pull request by kumarde**
Build error is fixed by #35. This fixes lints that do not properly check if they are CA certs before checking sub_cert specific tests.

### CODE REVIEW

**Comment by zakird** [comment]
fix tests and then you can merge.

File: lints/lint_sub_cert_crl_distribution_points_marked_critical.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ func (l *subCrlDistCrit) Initialize() error {
 
 func (l *subCrlDistCrit) CheckApplies(c *x509.Certificate) bool {
 	// Add conditions for application here
```
Comment (zakird) [inline lints/lint_sub_cert_crl_distribution_points_marked_critical.go]: what is sub cert.

File: lints/lint_sub_cert_certificate_policies_marked_critical.go [hunk 2]
```diff
@@ -20,7 +20,7 @@ func (l *subCertPolicyCrit) Initialize() error {
 }
 
 func (l *subCertPolicyCrit) CheckApplies(c *x509.Certificate) bool {
-	return util.IsExtInCert(c, util.CertPolicyOID)
+	return util.IsExtInCert(c, util.CertPolicyOID) && !util.IsCaCert(c)
```
Comment (zakird) [inline lints/lint_sub_cert_certificate_policies_marked_critical.go]: why is this only true for cas?

File: lints/lint_sub_cert_crl_distribution_points_does_not_contain_url.go [hunk 3]
```diff
@@ -24,7 +24,7 @@ func (l *subCRLDistNoURL) Initialize() error {
 
 func (l *subCRLDistNoURL) CheckApplies(c *x509.Certificate) bool {
 	// Add conditions for application here
-	return util.IsExtInCert(c, util.CrlDistOID)
+	return util.IsExtInCert(c, util.CrlDistOID) && !util.IsCaCert(c)
```
Comment (zakird) [inline lints/lint_sub_cert_crl_distribution_points_does_not_contain_url.go]: this should be true for things that aren't ca certs....

**Comment by zakird** [comment]
@kumarde we agreed to get rid of these checks i think. can you check and close and/or update?

**Comment by kumarde** [comment]
Closing based on a conversation we had out of band @zakird; I'll send a new PR with those changes.

**Pull request closed without merging**